.github/workflows/publish.yml

```diff
@@ -64,5 +112,6 @@ jobs:
         uses: softprops/action-gh-release@v1
         with:
           files: ./dist/*
+          tag_name: "v1.0.3-${{ needs.check-release.outputs.ghidra-ver }}"
```

Comment:
need to figure out a convention for the version number.  there is the version of the code within the project (v1.0.3) and the version of the corresponding Ghidra build to consider.

Reply:
IIRC the tags in the repo are the project version e.g. `v1.0.3`, the version pushed to PyPi is the combination e.g. `ghidra-stubs-9.2.2.1.0.3`

Reply:
I see now. The python package is generated using the current github tag ref and the ghidra version.  https://github.com/VDOO-Connected-Trust/ghidra-pyi-generator/blob/8a776b086d7e7c4a0029a418206f6613299382f3/.github/workflows/publish.yml#L35

How would you want to handle the github releases in the github workflow? The release step allows github to host a copy of the built package, but perhaps you want pypi to hold each build, and leave the release tag at v1.0.3 in github?

Reply:
I guess only updating PyPI in that case would be nice?

Reply:
Understanding now that the current release process for `ghidra-pyi-generator` only releases to github on a `push` of a tag `v*.*.*` and that is how you set the release version `${GITHUB_REF#refs/*/v}` (with pushing a new tag).  Do you want to keep that model? We could separate my proposed changes for automation into a new workflow file that would be responsible for publishing to pypi, checking weekly for new Ghidra releases, and able to be manually run on `workflow_dispatch`. The automated script would default use the "latest" release version tag from `ghidra-pyi-generator` (or it could be specified like the ghidra version).

If not, and you want to publish releases and also host the wheel and pypi package on github like I [have](https://github.com/clearbluejar/ghidra-pyi-generator/releases), we would need a way to define the "latest" version (such as a file in the repo) instead of relying on the push of a tag. The "latest" version file would be used to define

Reply:
I have no opinion on this. I'm not actively using this anymore.
Choose whichever you think will work best, and I'll be happy to approve it.

Reply:
Alright. I added [version.py](https://github.com/VDOO-Connected-Trust/ghidra-pyi-generator/pull/15/files#diff-bcc29717dbc0e48c1876977fe81f5515aecc40be0243f2802346fe68afc83f96) to store the current version ( I bumped it to 1.0.4.  This is read by the publish yaml now to generate the version like:

- v10.1.5.1.0.3
- v10.1.6.1.0.3 (a new ghidra release)
- v10.1.6.1.0.4 (change to ghidra-pyi-generator)

Once merged, it will build the latest release on the scheduled cron time specified and release like the latest release from my forked repo: https://github.com/clearbluejar/ghidra-pyi-generator/releases

I can't test the pypi publish step as I don't have access rights, but should work as intended. I don't do any checks for already existing pypi packages.

To build the previous Ghidra versions (10.1.4, 10.1.3, etc. ) you will need manually kick off each version you want to build. 
![image](https://user-images.githubusercontent.com/3752074/194388312-0bc3110e-6adf-4be3-a0ab-19d252fd066a.png) 

If you kickoff a a version to build other than 'latest' (which it sets on scheduled builds and new tags) it will always do the build step and publish a release. It only checks for existing versions on automated builds.

[From line 95 on the new publish.yaml](https://github.com/VDOO-Connected-Trust/ghidra-pyi-generator/pull/15/files#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R70):
```yaml
if: needs.check-release.outputs.already-exists == 'false' || (github.event_name == 'workflow_dispatch' && github.event.inputs.workflow-ghidra-ver != 'latest')
```

If publishing the old packages is to much, let me know. I can try to do it from my fork if needed.